contracts/USM.sol

```diff
@@ -72,7 +60,9 @@ contract USM is BufferedToken {
      */
     function burn(uint _usmToBurn) external returns (uint) {
         require(_usmToBurn >= MIN_BURN_AMOUNT, "1 USM minimum");
-        uint ethToSend = _burn(_usmToBurn);
+        // uint ethToSend = _burn(_usmToBurn);
+        uint ethToSend = usmToEth(_usmToBurn);
+        _burn(msg.sender, _usmToBurn);
         require(debtRatio() <= WAD, "Debt ratio too high");
```

Comment:
Removing `ethPool` in favour of the address balance has introduced an issue here.

Previously, our internal `_burn` made changes to the `ethPool` so that on this line, `debtRatio()` represents the future debt ratio after this transaction has completed.

Now, it won't because it uses the current balance, which will decrease a few lines later when the contract actually sends the ETH.

Reply:
Ha!  Yeah I almost wrote a comment about this yesterday - removing `ethPool` means we lose control of exactly at what point in the logic it changes, which can introduce problems like you describe.  May have to hand-calculate debt ratio based on the upcoming ETH amount.  Or, in [`usm_constproduct.py`](https://github.com/jacob-eliosoff/usm-stablecoin/blob/master/usm_constproduct.py#L241) I dealt with this by giving `debt_ratio()` optional `eth` and `usm` params so you could pass in values other than the current actual ones, that's one possibility.

Reply:
I think that you could move the code from the internal _mint and _burn functions to the public ones, and put the debt ratio require afterwards, as a post-condition.

Reply:
Turns out this is resolved by moving the require statements to be post-conditions! :)